auth0/resource_auth0_connection.go

```diff
@@ -789,8 +790,6 @@ func readConnection(ctx context.Context, d *schema.ResourceData, m interface{})
 		return diag.FromErr(err)
 	}
 
-	d.SetId(auth0.StringValue(connection.ID))
```

Comment:
From my understanding so far we should only set the ID of a resource when we create it, so no read method should set any ids as they are already set or if importing the `d.Id()` will be the id of the connection already.